flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSplitReader.java

```diff
@@ -120,6 +124,7 @@ private void checkSplitOrStartNext() throws IOException {
             currentSplitId = nextSplit.splitId();
 
             if (nextSplit.isSnapshotSplit()) {
+                metricRecorder.setInSnapshot(true);
```

Comment:
We can use ` splitState.isBinlogSplitState()` to record the metric in `MySqlRecordEmitter`